### PR TITLE
tests: function to test commands and unit tests for root.go

### LINKED ISSUE #70: cli: create tests for existing functionalities and commands
After #3 is done, we should test all the functionalities and commands we already have. The following list should summarize the different components we need to test:

- [x] Utils functions
- [x] Validation functions
- [x] `root` functions and general way to test command output
- [x] `close` command
- [x] `du` command
- [x] `info` command
- [ ] `list` command
- [x] `logs` command
- [x] `open` command
- [ ] `status` command
- [x] `ls` command

### PR DESCRIPTION
**Pull request by BrunoRosendo**
closes #70 

Review after #72

### CODE REVIEW

File: cmd/root_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+	"github.com/spf13/viper"
+)
+
+func testCmdRun(
+	t *testing.T,
+	cmd, serverPath, serverResponse string,
+	wantError bool, expectedMsgs []string,
+	args ...string,
+) {
```
Comment (mdonadoni) [inline cmd/root_test.go:30]: This function has quite a lot of parameters and it currently only support one endpoint, but we can improve this later when needed!

File: cmd/root_test.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	}
+
+	for _, test := range tests {
+		cmd := NewRootCmd()
+		f := cmd.Flags()
+		if test.hasToken {
+			f.String("access-token", test.token, "")
+		}
+		if test.hasServerURL {
+			viper.Set("server-url", test.serverURL)
```
Comment (mdonadoni) [inline cmd/root_test.go:129]: I think we need to call `viper.Reset()` at the end of each subtest, don't we?

**Comment by mdonadoni** [comment]
There is also a small typo in the commit title `cli: create unit tests tests for root.go` (the word `tests` is repeated two times). Should we also use `tests: ...` instead of `cli: ...`?

File: cmd/root_test.go [hunk 3]
```diff
@@ -63,3 +67,146 @@ func testCmdRun(
...
+		viperProp string
+		value     string
+	}{
+		{env: "REANA_SERVER_URL", viperProp: "server-url", value: "https://localhost:8080"},
+		{env: "REANA_ACCESS_TOKEN", viperProp: "access-token", value: "1234"},
+		{env: "REANA_WORKON", viperProp: "workflow", value: "workflow"},
+	}
+
+	for _, test := range tests {
+		err := os.Setenv(test.env, test.value)
```
[the remark this answers is not in the document]
Comment (BrunoRosendo) [reply cmd/root_test.go:166]: You're right. I'm doing the `t.Run(...)` changed in another PR, so I'm doing this there

**Review by mdonadoni — APPROVED** [review]
Nice job!

**Pull request merged**